webapp/channels/src/components/code_preview.tsx

```diff
@@ -20,130 +20,115 @@ type Props = {
-    }
+            setStatus({loading: false, success: false});
+        } else {
+            setCodeInfo({...codeInfo, code: '', lang: usedLanguage});
 
-    componentDidUpdate(prevProps: Props) {
-        if (this.props.fileUrl !== prevProps.fileUrl) {
-            this.getCode();
+            setStatus({...status, loading: true});
         }
+
+        setPrevFileUrl(fileUrl);
     }
 
-    getCode = async () => {
-        if (!this.state.lang || this.props.fileInfo.size > Constants.CODE_PREVIEW_MAX_FILE_SIZE) {
-            return;
-        }
-        try {
-            const data = await fetch(this.props.fileUrl);
-            const text = await data.text();
-            this.handleReceivedCode(text);
-        } catch (e) {
-            this.handleReceivedError();
-        }
-    };
+    useEffect(() => {
+        const handleReceivedCode = async (data: string | Node) => {
+            let code = data as string;
+            const Data = data as Node;
 
-    handleReceivedCode = async (data: string | Node) => {
-        let code = data as string;
-        const Data = data as Node;
-        if (Data.nodeName === '#document') {
-            code = new XMLSerializer().serializeToString(Data);
-        }
-        this.props.getContent?.(code);
-        this.setState({
-            code,
-            highlighted: await SyntaxHighlighting.highlight(this.state.lang, code),
-            loading: false,
-            success: true,
-        });
-    };
-
-    handleReceivedError = () => {
-        this.setState({loading: false, success: false});
-    };
-
-    static supports(fileInfo: FileInfo | LinkInfo) {
-        return Boolean(SyntaxHighlighting.getLanguageFromFileExtension(fileInfo.extension));
-    }
+            if (Data.nodeName === '#document') {
+                code = new XMLSerializer().serializeToString(Data);
+            }
 
-    render() {
-        if (this.state.loading) {
-            return (
-                <div className='view-image__loading'>
-                    <LoadingSpinner/>
-                </div>
-            );
-        }
+            getContent?.(code);
 
-        if (!this.state.success) {
-            return (
-                <FileInfoPreview
-                    fileInfo={this.props.fileInfo}
-                    fileUrl={this.props.fileUrl}
-                />
-            );
-        }
+            setCodeInfo({
+                ...codeInfo,
+                code,
+                highlighted: await SyntaxHighlighting.highlight(codeInfo.lang, code),
+            });
+
+            setStatus({loading: false, success: true});
+        };
 
-        const language = SyntaxHighlighting.getLanguageName(this.state.lang);
+        const handleReceivedError = () => {
+            setStatus({loading: false, success: false});
+        };
 
+        const getCode = async () => {
+            if (
+                !codeInfo.lang ||
+                fileInfo.size > Constants.CODE_PREVIEW_MAX_FILE_SIZE
+            ) {
```

Comment:
Hopefully, this condition has a higher level meaning we can extract to a constant outside of the effect. That way the effect will only run when actually needed (and we also remove completely the dependence on codeInfo). The cost is that we calculate the variable on every render, but should be a minimal cost.